tensorflow/contrib/rnn/python/kernel_tests/rnn_test.py

```diff
@@ -0,0 +1,355 @@
+"""Tests for rnn module."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import time
+import timeit
+
+import numpy as np
+from six.moves import xrange  # pylint: disable=redefined-builtin
+
+import tensorflow as tf
+from tensorflow.python.kernel_tests import rnn_test
+
+tf.contrib.rnn.Load()
+
+
+def _flatten(list_of_lists):
+  return [x for y in list_of_lists for x in y]
+
+
+class LSTMTest(tf.test.TestCase):
+
+  def setUp(self):
+    self._seed = 23489
+    np.random.seed(self._seed)
+
+  def _testLSTMBasicToCellBlockRNN(self, use_gpu, use_sequence_length, parallel_dw):
+    time_steps = 8
+    num_units = 3
+    num_proj = 4
+    input_size = 5
+    batch_size = 2
+
+    input_values = np.random.randn(time_steps, batch_size, input_size)
+
+    if use_sequence_length:
+      sequence_length = np.random.randint(0, time_steps, size=batch_size)
+    else:
+      sequence_length = None
+
+    ########### Step 1: Run BasicLSTMCell
+    initializer = tf.random_uniform_initializer(-0.01, 0.01, seed=self._seed)
+    basic_lstm_cell = tf.nn.rnn_cell.BasicLSTMCell(num_units)
+
+    (values_basic, state_value_basic, basic_grad_values,
+     basic_individual_grad_values,
+     basic_individual_var_grad_values) = rnn_test.run_static_rnn(
+         self, time_steps, batch_size, input_size, basic_lstm_cell,
+         sequence_length, input_values, use_gpu, initializer=initializer)
+
+    ########### Step 2: Run LSTMCellBlock
+    lstm_cell_block_cell = tf.contrib.rnn.LSTMCellBlock(num_units,
+                                                        parallel_dw=parallel_dw)
+
+    (values_block, state_value_block, block_grad_values,
+     block_individual_grad_values,
+     block_individual_var_grad_values) = rnn_test.run_static_rnn(
+         self, time_steps, batch_size, input_size, lstm_cell_block_cell,
+         sequence_length, input_values, use_gpu, initializer=initializer)
+
+    ######### Step 3: Comparisons
+    self.assertEqual(len(values_basic), len(values_block))
+    for (value_basic, value_block) in zip(values_basic, values_block):
+      self.assertAllClose(value_basic, value_block)
+
+    self.assertAllClose(basic_grad_values, block_grad_values)
+
+    self.assertEqual(len(basic_individual_grad_values),
+                     len(block_individual_grad_values))
+    self.assertEqual(len(basic_individual_var_grad_values),
+                     len(block_individual_var_grad_values))
+
+    for i, (a, b) in enumerate(zip(basic_individual_grad_values,
+                                   block_individual_grad_values)):
+      tf.logging.info("Comparing individual gradients iteration %d" % i)
+      self.assertAllClose(a, b)
+
+    for i, (a, b) in enumerate(reversed(zip(basic_individual_var_grad_values,
+                                   block_individual_var_grad_values))):
```

Comment:
sorry, what does "ws" refer to?

--
William Chan
Carnegie Mellon University
(650) 450-9455
williamchan.ca

On Fri, May 6, 2016 at 3:17 PM, ebrevdo <notifications@github.com> wrote:

> In tensorflow/contrib/rnn/python/kernel_tests/rnn_test.py
> <https://github.com/tensorflow/tensorflow/pull/2002#discussion_r62398353>:
>
> > +      self.assertAllClose(value_basic, value_block)
> > +
> > +    self.assertAllClose(basic_grad_values, block_grad_values)
> > +
> > +    self.assertEqual(len(basic_individual_grad_values),
> > +                     len(block_individual_grad_values))
> > +    self.assertEqual(len(basic_individual_var_grad_values),
> > +                     len(block_individual_var_grad_values))
> > +
> > +    for i, (a, b) in enumerate(zip(basic_individual_grad_values,
> > +                                   block_individual_grad_values)):
> > +      tf.logging.info("Comparing individual gradients iteration %d" % i)
> > +      self.assertAllClose(a, b)
> > +
> > +    for i, (a, b) in enumerate(reversed(zip(basic_individual_var_grad_values,
> > +                                   block_individual_var_grad_values))):
>
> is this ws correct?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/tensorflow/tensorflow/pull/2002/files/727e0c76bed212fdf240a221e577e1b1a86848d5#r62398353>
>